src/librustc_trans/glue.rs

```diff
@@ -235,56 +232,28 @@ impl<'tcx> DropGlueKind<'tcx> {
 
 fn get_drop_glue_core<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>,
                                 g: DropGlueKind<'tcx>) -> ValueRef {
-    debug!("make drop glue for {:?}", g);
     let g = g.map_ty(|t| get_drop_glue_type(ccx.tcx(), t));
-    debug!("drop glue type {:?}", g);
     match ccx.drop_glues().borrow().get(&g) {
-        Some(&glue) => return glue,
-        _ => { }
+        Some(&(glue, _)) => glue,
+        None => { bug!("Could not find drop glue for {:?} -- {} -- {}",
```

Comment:
maybe add a comment clarifying where this *ought* to have been created? is it as part of the "predefinition" code? (if so, that surprises me a **bit** since doesn't the "old trans" style stuff run before predefinition? or maybe not.)